mw_redis_cache.go

```diff
@@ -41,13 +44,42 @@ func (m *RedisCacheMiddleware) EnabledForSpec() bool {
 	return m.Spec.CacheOptions.EnableCache
 }
 
-func (m *RedisCacheMiddleware) CreateCheckSum(req *http.Request, keyName string) string {
+func (m *RedisCacheMiddleware) CreateCheckSum(req *http.Request, keyName string, regex string) (string, error) {
 	h := md5.New()
 	io.WriteString(h, req.Method)
 	io.WriteString(h, "-")
 	io.WriteString(h, req.URL.String())
+	if req.Method == "POST" {
+		if req.Body != nil {
+			bodyBytes, err := ioutil.ReadAll(req.Body)
+			defer req.Body.Close()
```

Comment:
`defer req.Body.Close()` should be called after `if err != nil {...}`

Reply:
Oh yeah. I will fix it.